x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportStartDataFrameAnalyticsAction.java

```diff
@@ -130,7 +130,7 @@ public void onFailure(Exception e) {
 
         // Validate config
         ActionListener<DataFrameAnalyticsConfig> configListener = ActionListener.wrap(
-            config -> DataFrameDataExtractorFactory.create(client, Collections.emptyMap(), config.getId(), config.getSource(),
+            config -> DataFrameDataExtractorFactory.create(client, Collections.emptyMap(), config,
```

Comment:
I moved to using the config as flip-flopping between source/dest seemed to confusing to me. I think it makes sense to create the same extractor here for validations as we do for the task..

Reply:
I'm not sure if that works. At the validation point we haven't created the `dest` index yet, right? So we have to test against the `source` index instead. Confusing as it may be, it helps leave things clean in case there are validation issues. Correct me if I'm wrong.

Reply:
Hmmmm, I see the point :/ but I am not going to lie, the switching between source and dest at different points in the call flow caused much consternation.